code/modules/reagents/chemistry/equilibrium.dm

```diff
@@ -0,0 +1,293 @@
+* This way all information is kept within one accessable object
+* equilibrium is a unique name as reaction is already too close to chemical_reaction
+* This is set up this way to reduce holder.dm bloat as well as reduce confusing list overhead
+* The crux of the fermimechanics are handled here
+* Instant reactions AREN'T handled here. See holder.dm
+*/
+/datum/equilibrium
+	///The chemical reaction that is presently being processed
+	var/datum/chemical_reaction/reaction 
+	///The location/reagents datum the processing is taking place
+	var/datum/reagents/holder 
+	///How much product we can make multiplied by the input recipe's products/required_reagents numerical values
+	var/multiplier = INFINITY
+	///The sum total of each of the product's numerical's values. This is so the addition/deletion is kept at the right values for multiple product reactions
+	var/product_ratio = 0
+	///The target volume the reaction is headed towards. This is updated every tick, so isn't the total value for the reaction, it's just a way to ensure we can't make more than is possible.
+	var/target_vol = INFINITY
+	///How much of the reaction has been made so far. Mostly used for subprocs, but it keeps track across the whole reaction and is added to every step.
+	var/reacted_vol = 0 
+	///If we're done with this reaction so that holder can clear it.
+	var/to_delete = FALSE 
+
+/* 
+* Creates and sets up a new equlibrium object
+* 
+* Arguments:
+* * input_reaction - the chemical_reaction datum that will be processed
+* * input_holder - the reagents datum that the output will be put into
+*/
+/datum/equilibrium/New(datum/chemical_reaction/input_reaction, datum/reagents/input_holder)
+	reaction = input_reaction
+	holder = input_holder
+	if(!holder || !reaction)
+		CRASH("A new [type] was set up, with incorrect/null input vars!")
+	if(!check_inital_conditions()) //If we're outside of the scope of the reaction vars
+		to_delete = TRUE
+		return
+	if(!length(reaction.results)) //Come back to and revise the affected reactions in the next PR, this is a placeholder fix.
+		holder.instant_react(reaction) //Even if this check fails, there's a backup - look inside of calculate_yield()
+		to_delete = TRUE
+		return
+	reaction.on_reaction(holder, multiplier)
+	SSblackbox.record_feedback("tally", "chemical_reaction", 1, "[reaction.type] attempts")
+	
+
+/datum/equilibrium/Destroy()
+	LAZYREMOVE(holder.reaction_list, src)
+	holder = null
+	reaction = null
+	return ..()
+
+/* 
+* Check to make sure our input vars are sensible - truncated version of check_reagent_properties() 
+* 
+* (as the setup in holder.dm checks for that already - this is a way to reduce calculations on New())
+* Don't call this unless you know what you're doing, this is an internal proc
+*/
+/datum/equilibrium/proc/check_inital_conditions()
+	//Make sure we have the right multipler for on_reaction()
+	for(var/B in reaction.required_reagents)
```

Comment:
Better variable names--avoid abbreviated names like `B` in general.

Reply:
Here and everywhere else by the way, you do this a lot.

Reply:
How should I do it when I'm doing 
```dm
for(var/r in list)
    var/datum/reagent/reagent = r
````
In this case, what would r be best called?

Reply:
`_reagent` is what I'd personally go with here.

Reply:
I think I got most of these, let me know if I mised anything.